core/src/main/scala/cats/mtl/MonadChronicle.scala

```diff
@@ -0,0 +1,29 @@
+package cats
+package mtl
+
+import cats.data.Ior
+
+trait MonadChronicle[F[_], E] extends Serializable {
+  val monad: Monad[F]
+
+  def dictate(c: E): F[Unit]
+
+  def disclose[A](c: E)(implicit M: Monoid[A]): F[A]
+
+  def confess[A](c: E): F[A]
+
+  def memento[A](ma: F[A]): F[Either[E, A]]
+
+  def absolve[A](a: A, ma: F[A]): F[A]
+
+  def condemn[A](ma: F[A]): F[A]
```

Comment:
With memento returning Ior this could be:

```scala
def condemn[A](ma: F[A]): F[A] =
  memento(ma).flatMap {
    case Left(e) => confess(e)
    case Right(a) => pure(a)
    case Both(_, a) => confess(e)
  }
```

Reply:
Yes! On the `Both` case we need `case Ior.Both(e, _) => confess(e)` :)